kubernetes/istio.go

```diff
@@ -167,20 +191,151 @@ func parseProxyStatus(statuses map[string][]byte) ([]*ProxyStatus, error) {
 	return fullStatus, nil
 }
 
-func parseRegistryServices(registries map[string][]byte) ([]*RegistryStatus, error) {
-	var fullRegistries []*RegistryStatus
+func parseRegistryServices(registries map[string][]byte) ([]*RegistryService, error) {
+	var fullRegistryServices []*RegistryService
 	for pilot, registry := range registries {
-		var rr []*RegistryStatus
+		var rr []*RegistryService
 		err := json.Unmarshal(registry, &rr)
 		if err != nil {
+			log.Errorf("Error parsing RegistryServices results: %s", err)
 			return nil, err
 		}
 		for _, r := range rr {
 			r.pilot = pilot
 		}
-		fullRegistries = append(fullRegistries, rr...)
+		fullRegistryServices = append(fullRegistryServices, rr...)
+	}
+	return fullRegistryServices, nil
+}
+
+func parseRegistryEndpoints(endpoints map[string][]byte) ([]*RegistryEndpoint, error) {
+	var fullRegistryEndpoints []*RegistryEndpoint
+	for pilot, endpoint := range endpoints {
+		var eps []*RegistryEndpoint
+		err := json.Unmarshal(endpoint, &eps)
+		if err != nil {
+			log.Errorf("Error parsing RegistryEndpoints results: %s", err)
+			return nil, err
+		}
+		for _, ep := range eps {
+			ep.pilot = pilot
```

Comment:
Yes, that's correct.
In the case of the "proxy-status" information we needed to iterate over all istiod pods to collect all the state.
On the registry endpoints perhaps is not necessary but I thought it could be useful to also bring this information for debugging purposes.

It's a good comment, I'm going to test this case to check that nothing is missed nor duplicated introduced.